src/main/java/ua/team3/carsharingservice/service/impl/RentalServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+package ua.team3.carsharingservice.service.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import java.time.LocalDate;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import ua.team3.carsharingservice.dto.RentalDto;
+import ua.team3.carsharingservice.dto.RentalRequestDto;
+import ua.team3.carsharingservice.exception.NoCarsAvailableException;
+import ua.team3.carsharingservice.exception.NotValidRentalDateException;
+import ua.team3.carsharingservice.exception.NotValidReturnDateException;
+import ua.team3.carsharingservice.exception.RentalAlreadyReturnedException;
+import ua.team3.carsharingservice.mapper.RentalMapper;
+import ua.team3.carsharingservice.model.Car;
+import ua.team3.carsharingservice.model.Rental;
+import ua.team3.carsharingservice.model.User;
+import ua.team3.carsharingservice.repository.CarRepository;
+import ua.team3.carsharingservice.repository.RentalRepository;
+import ua.team3.carsharingservice.service.RentalService;
+
+@Service
+@RequiredArgsConstructor
+public class RentalServiceImpl implements RentalService {
+    private static final int NUM_RENTAL_CARS = 1;
+    private final RentalRepository rentalRepository;
+    private final CarRepository carRepository;
+    private final RentalMapper rentalMapper;
+
+    @Override
+    public List<RentalDto> getAll(User user, Pageable pageable) {
+        return rentalRepository.findByUserId(user.getId(), pageable)
+                .stream()
+                .map(rentalMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public RentalDto getById(Long id, User user) {
+        Rental rental = getRentalByIdForUser(id, user);
+        return rentalMapper.toDto(rental);
+    }
+
+    @Override
+    @Transactional
+    public RentalDto create(RentalRequestDto rentalDto, User user) {
+        validateRentalPeriod(rentalDto);
+        validateRentalPermissionFor(user);
+
+        Car car = getCarById(rentalDto.getCarId());
+        decreaseInventoryInCar(car);
+
+        Rental rental = rentalMapper.toModel(rentalDto);
+        rental.setUser(user);
+        rental.setCar(car);
+        Rental savedRental = rentalRepository.save(rental);
+        return rentalMapper.toDto(savedRental);
+    }
+
+    private void decreaseInventoryInCar(Car car) {
+        int availableCarInventory = car.getInventory();
+        if (availableCarInventory < NUM_RENTAL_CARS) {
+            throw new NoCarsAvailableException(
+                    "There are no available cars of this type, please choose another"
+            );
+        }
+        car.setInventory(availableCarInventory - NUM_RENTAL_CARS);
```

Comment:
```suggestion
        car.setInventory(--availableCarInventory);
```

Reply:
I think it will be more beautiful, but maybe I didn't take something into consideration